### PR TITLE
Update gazelle to v0.20.0

### LINKED ISSUE #501: Update gazelle
We're using an old version which lacks some nice features such as support for the new go module system.  This will be an issue when adding dependencies on go code which uses the new modules.

Since the new version supports go.mod and go.sum, I also think we should switch to autogenerating all our gazelle rules using `gazelle update-repos` once we do this.  This will make go dependency management require significantly less manual effort.

### PR DESCRIPTION
**Pull request by cryslith**
Also upgrade the prometheus golang client dependencies with it,
since that's what I needed the new gazelle version for in the first place.

Closes #501.

---

Checklist:

 - [x] I have split up this change into one or more appropriately-delineated commits.
 - [x] The first line of each commit is of the form "[component]: do something"
 - [x] I have written a complete, multi-line commit message for each commit.
 - [x] I have formatted any Go code that I have changed with gofmt.
 - [x] I have written or updated appropriate documentation to cover this change.
 - [x] I have confirmed that this change is covered by at least one appropriate test run by CI.
 - [x] If my change includes new or modified functionality, I have tested that the changes work as expected.
 - [x] I have assigned this issue to an appropriate reviewer. (Choose @celskeggs if you are not otherwise certain.)
 - [x] I consider my PR complete and ready to be merged without my further input, assuming that it passes CI and code review.
 - [x] My changes have passed CI, including an automatic Jenkins deploy.
 - [x] My changes have passed code review.

### CODE REVIEW

File: platform/cri-o/build.patch [hunk 1]
```diff
@@ -162,9 +162,5 @@ index 3a0a989..9e0efbe 100644
      importmap = "github.com/cri-o/cri-o/vendor/golang.org/x/crypto/ed25519",
      importpath = "golang.org/x/crypto/ed25519",
      visibility = ["//visibility:public"],
--    deps = [
--        "//vendor/crypto/ed25519:go_default_library",
--        "//vendor/golang.org/x/crypto/ed25519/internal/edwards25519:go_default_library",
--    ],
-+    deps = ["//vendor/golang.org/x/crypto/ed25519/internal/edwards25519:go_default_library"],
```
Comment (celskeggs) [inline platform/cri-o/build.patch:169]: I'm curious why this change is part of this commit. Could this perhaps be explained in the commit message?

Comment (cryslith) [reply platform/cri-o/build.patch:169]: Well, it seems that gazelle started doing what we want; I'm not sure why.  Note that my change replaces `change X to Y` with `leave Y unchanged`; i.e. gazelle has stopped incorrectly producing `X` and is now correctly producing `Y` instead, so we don't need to do anything.

Comment (celskeggs) [reply platform/cri-o/build.patch:169]: Hmm, okay. That makes sense.

**Comment by celskeggs** [comment]
Waiting on the last two checklist items to merge.
